drivers/all_neighbors_system_dashboard/app/models/all_neighbors_system_dashboard/enrollment_attribute_calculations.rb

```diff
@@ -0,0 +1,188 @@
+        15 => 447, # Referral to Other PH project/unit/resource opening
+        # 16 => 'Referral to emergency assistance/flex fund/furniture assistance',
+        17 => 1114, # Referral to Emergency Housing Voucher (EHV)
+        # 18 => # Referral to a Housing Stability Voucher
+      }.freeze
+
+      def household_type(enrollment)
+        return 'Children Only' if enrollment.children_only?
+        # The enrollment isn't children only, but contains at least one child
+        return 'Adults and Children' if enrollment.other_clients_under_18.positive? ||
+          (enrollment.age.present? && enrollment.age < 18)
+
+        'Adults Only'
+      end
+
+      def prior_living_situation_category(enrollment)
+        case enrollment.living_situation
+        when 30, 17, 37, 8, 9, 99
+          'Unknown'
+        when 16
+          'Unsheltered'
+        when 1, 18, 2
+          'Sheltered'
+        when 15, 6, 7, 25, 4, 5
+          'Institutional'
+        when 29, 14, 32, 13, 27, 12, 22, 35, 36, 23, 26, 28, 19, 3, 31, 33, 34, 10, 20, 21, 11
+          'Housed'
+        when 24
+          'Deceased'
+        else
+          'ERROR'
+        end
+      end
+
+      def exit_date(filter, enrollment)
+        date = enrollment.exit_date
+        return nil if date.present? && date > filter.end_date
+
+        date
+      end
+
+      def adjusted_exit_date(filter, enrollment)
+        date = enrollment.exit_date
+        return filter.end_date if date.blank?
+        return nil if date.present? && date > filter.end_date
+
+        date
+      end
+
+      def exit_type(enrollment)
+        case enrollment.destination
+        when nil
+          nil
+        when 26, 11, 21, 3, 10, 28, 20, 19, 22, 23, 31, 33, 34
+          'Permanent'
+        when 6, 24, 15
+          'Excludable'
+        when 8, 9, 99, 30, 17
+          'Unknown'
+        else
+          'Non-Permanent'
+        end
+      end
+
+      def relationship(enrollment)
+        case enrollment.relationship_to_ho_h
+        when 1
+          'SL'
+        when 2
+          'DC'
+        when 3
+          'SP'
+        when 4
+          'F'
+        when 5
+          'O'
+        end
+      end
+
+      def gender(enrollment)
+        HudUtility.gender(enrollment.client.gender_binary)
+      end
+
+      def primary_race(enrollment)
+        HudUtility.race(enrollment.client.pit_race, multi_racial: true)
+      end
+
+      def race_list(enrollment)
+        enrollment.client.race_description
+      end
+
+      def ce_infos_for_batch(filter, batch)
+        household_enrollments = GrdaWarehouse::ServiceHistoryEnrollment.
+          preload(:enrollment).
+          entry.
+          where(project_id: filter.secondary_project_ids).
+          where(household_id: batch.map(&:household_id), data_source_id: batch.map(&:data_source_id)).
+          where(she_t[:exit_date].eq(nil).or(she_t[:exit_date].gteq(filter.start_date))).
```

Comment:
Why aren't we using one of the overlapping date range scopes?  This probably fine, but seems suspicious.